code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/CacheManager.java

```diff
@@ -0,0 +1,112 @@
+package io.cattle.platform.allocator.service;
+
+import io.cattle.platform.core.constants.HostConstants;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.util.DataAccessor;
+import io.cattle.platform.util.type.CollectionUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CacheManager
```

Comment:
Note to self: synchronizing on the methods herein would introduce a global block (across all environments).